notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+
+
+```python
+# HIDDEN
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from IPython.display import Image
+%matplotlib inline
+sns.set()
+sns.set_context('talk')
+```
+
+# Least Squares Regression as a Projection
+
+Recall that we found the optimal coefficients for linear models by optimizing their cost functions with gradient descent.
+
+We also mentioned that least squares linear regression can be solved analytically. While gradient descent is practical, this geometric perspective will provide a deeper understanding of linear regression.
+
+## Case Study
+
+We've been tasked with finding a good linear model and a rigorous argument for how we know that our model is sound, using the following data:
+
+| x | y |
+| - |:-|
+| 3 | 2 |
+| 0 | 1 |
+| -1 | -2 |
+
+
+```python
+# HIDDEN
+data = pd.DataFrame(
+    [
+        [3,2],
+        [0,1],
+        [-1,-2]
+    ],
+    columns=['x', 'y']
+)
+
+sns.regplot(x='x', y='y', data=data, ci=None)
+```
+
+
+
+
+    <matplotlib.axes._subplots.AxesSubplot at 0x1111584e0>
+
+
+
+
+![png](linear_projection_files/linear_projection_6_1.png)
+
+
+Assume that the best model is one with the least error, and that the least squared error is an acceptable measure.
+
+Recall that a simple linear regression model follows the form:
+
+$$
+\begin{aligned}
+f_\hat{\theta} (x_i) 
+&= \hat{\theta_0} + \hat{\theta_1} x_i \\
+\end{aligned}
+$$
+
+Our goal is to find the $\hat{\theta}$ that results in the line with the least squared error $L(\hat{\theta}, \vec{x}, \vec{y})$:
+
+$$
+\begin{aligned}
+L(\hat{\theta}, \vec{x}, \vec{y})
+&= \sum_{i = 1}^{n}(y_i - f_\hat{\theta} (x_i))^2\\
+&= \sum_{i = 1}^{n}(y_i - \hat{\theta_0} - \hat{\theta_1} x_i)^2\\
+&= \sum_{i = 1}^{n}(y_i - \begin{bmatrix} 1 & x_i \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix} ) ^2
+\end{aligned}
+$$
+
+In our data set, n = 3, so let's expand out this equation by breaking up the summation.
+
+$$
+\begin{aligned}
+L(\hat{\theta}, \vec{x}, \vec{y})
+&=
+(y_1 - \begin{bmatrix} 1 & x_1 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2  \\
+&+
+(y_2 - \begin{bmatrix} 1 & x_2 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2 \\
+&+
+(y_3 - \begin{bmatrix} 1 & x_3 \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix})^2 \\
+\end{aligned}
+$$
+
+Notice that our loss function is a sum of squares. Now, notice that the *L2*-norm for a vector has a similar form: 
+
+$$\Vert \vec{v} \Vert = \sqrt{v_1^2 + v_2^2 + \dots + v_n^2}$$ 
+
+So if we can express each $y_i - \begin{bmatrix} 1 & x_i \end{bmatrix}
+\begin{bmatrix} 
+     \hat{\theta_0} \\
+     \hat{\theta_1}
+\end{bmatrix}$ as a vector, we can express our loss as the *L2*-norm, squared:
```

Comment:
express our loss as the the *L2* norm of a vector, squared

Reply:
hmm i feel like this jump in general is pretty confusing :/ i don't really get it myself, how you changed it to the L2 norm of something

Reply:
does this help? 

![image](https://user-images.githubusercontent.com/18621425/41825557-8404300a-77d6-11e8-867f-0937357b375e.png)

Reply:
yep nice!